update mom6 to its main branch 20220602 commit, Allow wave grid to be different than ocean/ice grid #1279, Update rt.conf to compile all suites with Intel in debug mode #1230

jiandewang opened the pull request:
# PR Checklist

- [x] This PR is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR. Please consult the ufs-weather-model [wiki](https://github.com/ufs-community/ufs-weather-model/wiki/Making-code-changes-in-the-UFS-weather-model-and-its-subcomponents) if you are unsure how to do this.

- [x] This PR has been tested using a branch which is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR

- [x] An Issue describing the work contained in this PR has been created either in the subcomponent(s) or in the ufs-weather-model. The Issue should be created in the repository that is most relevant to the changes in contained in the PR. The Issue and the dependent sub-component PR 
are specified below.

- [ ] Results for one or more of the regression tests change and the reasons for the changes are understood and explained below.

- [ ] New or updated input data is required by this PR. If checked, please work with the code managers to update input data sets on all platforms.

## Instructions: All subsequent sections of text should be filled in as appropriate.

The information provided below allows the code managers to understand the changes relevant to this PR, whether those changes are in the ufs-weather-model repository or in a subcomponent repository. Ufs-weather-model code managers will use the information provided to add any applicable labels, assign reviewers and place it in the Commit Queue. Once the PR is in the Commit Queue, it is the PR owner's responsibility to keep the PR up-to-date with the develop branch of ufs-weather-model. 

## Description

Provide a detailed description of what this PR does. What bug does it fix, or what feature does it add? Is a change of answers expected from this PR? Are any library updates included in this PR (modulefiles etc.)?

MOM6 main branch has been updated on 20220602, need to update dev/emc correspondingly.
### Issue(s) addressed #1255 and MOM6 issue # https://github.com/NOAA-EMC/MOM6/issues/91
There will be no answer change in UFS

Link the issues to be closed with this PR, whether in this repository, or in another repository.
(Remember, issues must always be created before starting work on a PR branch!) 
- fixes MOM6 issue https://github.com/NOAA-EMC/MOM6/issues/91
- fixes UFS issue https://github.com/ufs-community/ufs-weather-model/issues/1255
-fixes #1278 

## Testing

How were these changes tested? What compilers / HPCs was it tested with? Are the changes covered by regression tests? (If not, why? Do new tests need to be added?) Have regression tests and unit tests (utests) been run? On which platforms and with which compilers? (Note that unit tests can only be run on tier-1 platforms)

- [x] hera.intel
- [x] hera.gnu
- [x] orion.intel
- [x] cheyenne.intel 
- [x] cheyenne.gnu
- [x] gaea.intel 
- [x] jet.intel
- [x] wcoss2.intel
- [x] wcoss_cray
- [x] wcoss_dell_p3
- [ ] opnReqTest for newly added/changed feature
- [x] CI [2537963438](https://github.com/ufs-community/ufs-weather-model/actions/runs/2537963438)

## Dependencies

- waiting on https://github.com/NOAA-EMC/MOM6/pull/92
- waiting on https://github.com/NOAA-EMC/CMEPS/pull/68

DeniseWorthen commented:
@jiandewang Please fill out the template w/ the additional info. Thanks.

jiandewang commented:
> @jiandewang Please fill out the template w/ the additional info. Thanks.

added related issue#

DeniseWorthen commented:
Please fill out the top, where you note whether the baselines change etc.

jiandewang commented:
> Please fill out the top, where you note whether the baselines change etc.

added no BL change information

DeniseWorthen commented:
Please fill out the PR Checklist.

jiandewang commented:
> Please fill out the PR Checklist.

marked in PR checklist

DeniseWorthen commented:
@jiandewang Thanks for adding the PR template info. I also added the PR dependency at the end (linking to your MOM6 PR). In the future, the EPIC team will be checking for a complete template so keep that in mind.

jiandewang commented:
> @jiandewang Thanks for adding the PR template info. I also added the PR dependency at the end (linking to your MOM6 PR). In the future, the EPIC team will be checking for a complete template so keep that in mind.

thanks @DeniseWorthen

jkbk2004 commented:
@jiandewang @climbfuji can we combine in #1230 to this PR? so that we can merge together.

jiandewang commented:
@jkbk2004 I think that will be fine

jiandewang commented:
I will wait for @climbfuji to update his UFS branch, then create PR to his branch for him to merge

jiandewang commented:
@jkbk2004 @junwang-noaa @climbfuji : Dom's PR has been merged to mine.

jkbk2004 commented:
@DeniseWorthen So the PR to combine the cmeps change of #1279 to this Jiande's branch?

jiandewang commented:
I am having conflict here in my sync from develop branch, which one shall I choose from below ?
138 <<<<<<< HEAD
    139 COMPILE | -DAPP=ATM -DDEBUG=ON -DCCPP_SUITES=FV3_RAP,FV3_RAP_cires_ugwp,FV3_RAP_unified_ugwp,FV3_RAP_flake -D32BIT=ON             | + wcoss_dell_p3         wcoss_cray              | fv3 |
    140 =======
    141 COMPILE | -DAPP=ATM -DDEBUG=ON -DCCPP_SUITES=FV3_RAP,FV3_RAP_cires_ugwp,FV3_RAP_unified_ugwp,FV3_RAP_flake,FV3_HRRR -D32BIT=ON    |                                                 | fv3 |
    142 >>>>>>> upstream/develop

jiandewang commented:
> @DeniseWorthen So the PR to combine the cmeps change of #1279 to this Jiande's branch?

@DeniseWorthen please wait until I solve the conflict before create PR to my branch

DeniseWorthen commented:
@jiandewang I've just updated to the latest develop in my branch. You should be able to just point to my branch in your .gitmodules once you're ready.

jiandewang commented:
> @jiandewang I've just updated to the latest develop in my branch. You should be able to just point to my branch in your .gitmodules once you're ready.

@DeniseWorthen  I just solved the conflict and updated my branch, not quite understand on what you said "point to my branch in .gitmodule".

jiandewang commented:
> > @jiandewang I've just updated to the latest develop in my branch. You should be able to just point to my branch in your .gitmodules once you're ready.
> 
> @DeniseWorthen I just solved the conflict and updated my branch, not quite understand on what you said "point to my branch in .gitmodule".

do you mean I change in my branch
url = https://github.com/DeniseWorthen/CMEPS
  branch = feature/mapwav

DeniseWorthen commented:
@jiandewang Yes, that is what I meant. 

I can make a PR if you'd prefer.

jiandewang commented:
> @jiandewang Yes, that is what I meant.
> 
> I can make a PR if you'd prefer.

yes a PR to my branch will be a clean one

DeniseWorthen commented:
@jiandewang done

jiandewang commented:
> @jiandewang done

@DeniseWorthen merged, thanks. As usual I will do all wcoss jobs

jiandewang commented:
@junwang-noaa @jkbk2004 : now all the PR combinations are done, let me know when it's the time for me to launch jobs on wcoss

jkbk2004 commented:
@jiandewang @DeniseWorthen let me start running Orion first. Once it goes ok, then we can submit across machines.

DeniseWorthen commented:
@jkbk2004 I did not have time to test on Gnu, so I would perhaps suggest starting one of the Gnu RTs also.

jkbk2004 commented:
> @jkbk2004 I did not have time to test on Gnu, so I would perhaps suggest starting one of the Gnu RTs also.

@DeniseWorthen Sure! I will follow up.

jiandewang commented:
I think FV3 is not up to date here, let me update that first

jkbk2004 commented:
@jiandewang current fv3atm hash is at 9771e

jiandewang commented:
> @jiandewang current fv3atm hash is at 9771e

@jkbk2004 just updated FV3

BrianCurtis-NOAA commented:
***Automated RT Failure Notification***
Machine: orion
Compiler: intel
Job: RT
[RT] Repo location: /work/noaa/nems/emc.nemspara/autort/pr/959824692/20220617131518/ufs-weather-model
[RT] Error: Test regional_3km 046 failed in check_result failed 
[RT] Error: Test regional_3km 046 failed in run_test failed 
[RT] Error: Test regional_3km_decomp 047 failed in check_result failed 
[RT] Error: Test regional_3km_decomp 047 failed in run_test failed 
[RT] Error: Test rap_control 048 failed in check_result failed 
[RT] Error: Test rap_control 048 failed in run_test failed 
[RT] Error: Test rap_rrtmgp 049 failed in check_result failed 
[RT] Error: Test rap_rrtmgp 049 failed in run_test failed 
[RT] Error: Test regional_spp_sppt_shum_skeb 050 failed in check_result failed 
[RT] Error: Test regional_spp_sppt_shum_skeb 050 failed in run_test failed 
[RT] Error: Test rap_decomp 051 failed in check_result failed 
[RT] Error: Test rap_decomp 051 failed in run_test failed 
[RT] Error: Test rap_2threads 052 failed in check_result failed 
[RT] Error: Test rap_2threads 052 failed in run_test failed 
[RT] Error: Test rap_sfcdiff 054 failed in check_result failed 
[RT] Error: Test rap_sfcdiff 054 failed in run_test failed 
[RT] Error: Test rap_sfcdiff_decomp 055 failed in check_result failed 
[RT] Error: Test rap_sfcdiff_decomp 055 failed in run_test failed 
[RT] Error: Test rrfs_conus13km_hrrr_warm 064 failed in check_result failed 
[RT] Error: Test rrfs_conus13km_hrrr_warm 064 failed in run_test failed 
[RT] Error: Test rrfs_conus13km_radar_tten_warm 065 failed in check_result failed 
[RT] Error: Test rrfs_conus13km_radar_tten_warm 065 failed in run_test failed 
[RT] Error: Test rrfs_smoke_conus13km_hrrr_warm 066 failed in check_result failed 
[RT] Error: Test rrfs_smoke_conus13km_hrrr_warm 066 failed in run_test failed 
[RT] Error: Test rap_control_debug 090 failed in check_result failed 
[RT] Error: Test rap_control_debug 090 failed in run_test failed 
[RT] Error: Test rap_unified_drag_suite_debug 092 failed in check_result failed 
[RT] Error: Test rap_unified_drag_suite_debug 092 failed in run_test failed 
[RT] Error: Test rap_diag_debug 093 failed in check_result failed 
[RT] Error: Test rap_diag_debug 093 failed in run_test failed 
[RT] Error: Test rap_cires_ugwp_debug 094 failed in check_result failed 
[RT] Error: Test rap_cires_ugwp_debug 094 failed in run_test failed 
[RT] Error: Test rap_unified_ugwp_debug 095 failed in check_result failed 
[RT] Error: Test rap_unified_ugwp_debug 095 failed in run_test failed 
[RT] Error: Test rap_lndp_debug 096 failed in check_result failed 
[RT] Error: Test rap_lndp_debug 096 failed in run_test failed 
[RT] Error: Test rap_flake_debug 097 failed in check_result failed 
[RT] Error: Test rap_flake_debug 097 failed in run_test failed 
[RT] Error: Test rap_progcld_thompson_debug 098 failed in check_result failed 
[RT] Error: Test rap_progcld_thompson_debug 098 failed in run_test failed 
[RT] Error: Test rap_noah_debug 099 failed in run_test failed 
[RT] Error: Test rap_rrtmgp_debug 100 failed in run_test failed 
[RT] Error: Test rap_sfcdiff_debug 101 failed in run_test failed 
[RT] Error: Test rap_noah_sfcdiff_cires_ugwp_debug 102 failed in run_test failed 
[RT] Error: Test rrfs_v1beta_debug 103 failed in run_test failed 
[RT] Error: Test hafs_regional_atm 105 failed in check_result failed 
[RT] Error: Test hafs_regional_atm 105 failed in run_test failed 
[RT] Error: Test hafs_regional_atm_thompson_gfdlsf 106 failed in check_result failed 
[RT] Error: Test hafs_regional_atm_thompson_gfdlsf 106 failed in run_test failed 
[RT] Error: Test hafs_regional_atm_ocn 107 failed in check_result failed 
[RT] Error: Test hafs_regional_atm_ocn 107 failed in run_test failed 
[RT] Error: Test hafs_regional_atm_wav 108 failed in check_result failed 
[RT] Error: Test hafs_regional_atm_wav 108 failed in run_test failed 
[RT] Error: Test hafs_regional_atm_ocn_wav 109 failed in check_result failed 
[RT] Error: Test hafs_regional_atm_ocn_wav 109 failed in run_test failed 
[RT] Error: Test hafs_regional_1nest_atm 110 failed in check_result failed 
[RT] Error: Test hafs_regional_1nest_atm 110 failed in run_test failed 
[RT] Error: Test hafs_regional_telescopic_2nests_atm 111 failed in check_result failed 
[RT] Error: Test hafs_regional_telescopic_2nests_atm 111 failed in run_test failed 
[RT] Error: Test hafs_regional_specified_moving_1nest_atm 114 failed in check_result failed 
[RT] Error: Test hafs_regional_specified_moving_1nest_atm 114 failed in run_test failed 
[RT] Error: Test hafs_regional_storm_following_1nest_atm 115 failed in check_result failed 
[RT] Error: Test hafs_regional_storm_following_1nest_atm 115 failed in run_test failed 
[RT] Error: Test hafs_regional_storm_following_1nest_atm_ocn 116 failed in check_result failed 
[RT] Error: Test hafs_regional_storm_following_1nest_atm_ocn 116 failed in run_test failed 
[RT] Error: Test hafs_regional_storm_following_1nest_atm_ocn_wav 117 failed in check_result failed 
[RT] Error: Test hafs_regional_storm_following_1nest_atm_ocn_wav 117 failed in run_test failed 
[RT] Error: Test hafs_regional_docn 119 failed in check_result failed 
[RT] Error: Test hafs_regional_docn 119 failed in run_test failed 
[RT] Error: Test hafs_regional_docn_oisst 120 failed in check_result failed 
[RT] Error: Test hafs_regional_docn_oisst 120 failed in run_test failed 
[RT] Error: Test compile_001 failed in run_compile failed 
[RT] Error: Test compile_003 failed in run_compile failed 
Please make changes and add the following label back: orion-intel-RT

DeniseWorthen commented:
Orion failure caused by not having FV3 up-to-date before labels, right?

jkbk2004 commented:
@jiandewang I think I have to resubmit jobs on orion: the failure notes fore before fv3atm updates.

jkbk2004 commented:
> @jiandewang I think I have to resubmit jobs on orion: the failure notes fore before fv3atm updates.

I think so. I will submit again

jiandewang commented:
> @jiandewang I think I have to resubmit jobs on orion: the failure notes fore before fv3atm updates.

it is 9771e5417af138287bd9c1628063106656347d65 FV3 (heads/develop) now, I updated ~1hr ago

BrianCurtis-NOAA commented:
***Automated RT Failure Notification***
Machine: orion
Compiler: intel
Job: RT
[RT] Repo location: /work/noaa/nems/emc.nemspara/autort/pr/959824692/20220617160013/ufs-weather-model
[RT] Error: Test rap_noah_debug 099 failed in run_test failed 
[RT] Error: Test rap_rrtmgp_debug 100 failed in run_test failed 
[RT] Error: Test rap_sfcdiff_debug 101 failed in run_test failed 
[RT] Error: Test rap_noah_sfcdiff_cires_ugwp_debug 102 failed in run_test failed 
[RT] Error: Test rrfs_v1beta_debug 103 failed in run_test failed 
Please make changes and add the following label back: orion-intel-RT

jkbk2004 commented:
@jiandewang @DeniseWorthen sounds like slurm issue with those orion.intel failures. I will manually submit those.

jiandewang commented:
can some tell me right now which machine is for development on wcoss-2 ?

BrianCurtis-NOAA commented:
***Automated RT Failure Notification***
Machine: jet
Compiler: intel
Job: RT
[RT] Repo location: /lfs4/HFIP/h-nems/emc.nemspara/autort/pr/959824692/20220617210008/ufs-weather-model
[RT] Error: Test rap_noah_debug 095 failed in run_test failed 
[RT] Error: Test rap_rrtmgp_debug 096 failed in run_test failed 
[RT] Error: Test rap_sfcdiff_debug 097 failed in run_test failed 
[RT] Error: Test rap_noah_sfcdiff_cires_ugwp_debug 098 failed in run_test failed 
[RT] Error: Test rrfs_v1beta_debug 099 failed in run_test failed 
Please make changes and add the following label back: jet-intel-RT

BrianCurtis-NOAA commented:
***Automated RT Failure Notification***
Machine: gaea
Compiler: intel
Job: RT
[RT] Repo location: /lustre/f2/pdata/ncep/emc.nemspara/autort/pr/959824692/20220617210006/ufs-weather-model
[RT] Error: Test rap_noah_debug 098 failed in run_test failed 
[RT] Error: Test rap_rrtmgp_debug 099 failed in run_test failed 
[RT] Error: Test rap_sfcdiff_debug 100 failed in run_test failed 
[RT] Error: Test rap_noah_sfcdiff_cires_ugwp_debug 101 failed in run_test failed 
[RT] Error: Test rrfs_v1beta_debug 102 failed in run_test failed 
Please make changes and add the following label back: gaea-intel-RT

BrianCurtis-NOAA commented:
***Automated RT Failure Notification***
Machine: hera
Compiler: intel
Job: RT
[RT] Repo location: /scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/959824692/20220617210013/ufs-weather-model
[RT] Error: Test rap_noah_debug 099 failed in run_test failed 
[RT] Error: Test rap_rrtmgp_debug 100 failed in run_test failed 
[RT] Error: Test rap_sfcdiff_debug 101 failed in run_test failed 
[RT] Error: Test rap_noah_sfcdiff_cires_ugwp_debug 102 failed in run_test failed 
[RT] Error: Test rrfs_v1beta_debug 103 failed in run_test failed 
Please make changes and add the following label back: hera-intel-RT

jkbk2004 commented:
I checked rap_noah_debug failed case. Develop branch runs ok but it crashes with the changes with the PR:
111:An error occurred in ccpp_physics_init
111:Invalid suite FV3_RAP_noah

jiandewang commented:
138 <<<<<<< HEAD
139 COMPILE | -DAPP=ATM -DDEBUG=ON -DCCPP_SUITES=FV3_RAP,FV3_RAP_cires_ugwp,FV3_RAP_unified_ugwp,FV3_RAP_flake -D32BIT=ON | + wcoss_dell_p3 wcoss_cray | fv3 |
140 =======
141 COMPILE | -DAPP=ATM -DDEBUG=ON -DCCPP_SUITES=FV3_RAP,FV3_RAP_cires_ugwp,FV3_RAP_unified_ugwp,FV3_RAP_flake,FV3_HRRR -D32BIT=ON | | fv3 |
142 >>>>>>> upstream/develop

this is what I got when merging @climbfuji branch, I used the develop one in my push, I guess Dom didn't get chance to test for this before he created PR to my branch

jkbk2004 commented:
> 138 <<<<<<< HEAD 139 COMPILE | -DAPP=ATM -DDEBUG=ON -DCCPP_SUITES=FV3_RAP,FV3_RAP_cires_ugwp,FV3_RAP_unified_ugwp,FV3_RAP_flake -D32BIT=ON | + wcoss_dell_p3 wcoss_cray | fv3 | 140 ======= 141 COMPILE | -DAPP=ATM -DDEBUG=ON -DCCPP_SUITES=FV3_RAP,FV3_RAP_cires_ugwp,FV3_RAP_unified_ugwp,FV3_RAP_flake,FV3_HRRR -D32BIT=ON | | fv3 | 142 >>>>>>> upstream/develop
> 
> this is what I got when merging @climbfuji branch, I used the develop one in my push, I guess Dom didn't get chance to test for this before he created PR to my branch

@jiandewang @climbfuji I think we can revert line #139. gaea is down today but I tested ok with the reverted line #139.

jkbk2004 commented:
> 138 <<<<<<< HEAD 139 COMPILE | -DAPP=ATM -DDEBUG=ON -DCCPP_SUITES=FV3_RAP,FV3_RAP_cires_ugwp,FV3_RAP_unified_ugwp,FV3_RAP_flake -D32BIT=ON | + wcoss_dell_p3 wcoss_cray | fv3 | 140 ======= 141 COMPILE | -DAPP=ATM -DDEBUG=ON -DCCPP_SUITES=FV3_RAP,FV3_RAP_cires_ugwp,FV3_RAP_unified_ugwp,FV3_RAP_flake,FV3_HRRR -D32BIT=ON | | fv3 | 142 >>>>>>> upstream/develop
> 
> this is what I got when merging @climbfuji branch, I used the develop one in my push, I guess Dom didn't get chance to test for this before he created PR to my branch

@jiandewang @climbfuji I think we can revert the line 139. geae is down today but I tested ok with the reverted line 139.

jiandewang commented:
> > 138 <<<<<<< HEAD 139 COMPILE | -DAPP=ATM -DDEBUG=ON -DCCPP_SUITES=FV3_RAP,FV3_RAP_cires_ugwp,FV3_RAP_unified_ugwp,FV3_RAP_flake -D32BIT=ON | + wcoss_dell_p3 wcoss_cray | fv3 | 140 ======= 141 COMPILE | -DAPP=ATM -DDEBUG=ON -DCCPP_SUITES=FV3_RAP,FV3_RAP_cires_ugwp,FV3_RAP_unified_ugwp,FV3_RAP_flake,FV3_HRRR -D32BIT=ON | | fv3 | 142 >>>>>>> upstream/develop
> > this is what I got when merging @climbfuji branch, I used the develop one in my push, I guess Dom didn't get chance to test for this before he created PR to my branch
> 
> @jiandewang @climbfuji I think we can revert the line 139. geae is down today but I tested ok with the reverted line 139.

Both Sam and Dom modified that compiling line, that's why I got conflict. @climbfuji can you confirm on what @jkbk2004 suggested in the above ?

jiandewang commented:
> > > 138 <<<<<<< HEAD 139 COMPILE | -DAPP=ATM -DDEBUG=ON -DCCPP_SUITES=FV3_RAP,FV3_RAP_cires_ugwp,FV3_RAP_unified_ugwp,FV3_RAP_flake -D32BIT=ON | + wcoss_dell_p3 wcoss_cray | fv3 | 140 ======= 141 COMPILE | -DAPP=ATM -DDEBUG=ON -DCCPP_SUITES=FV3_RAP,FV3_RAP_cires_ugwp,FV3_RAP_unified_ugwp,FV3_RAP_flake,FV3_HRRR -D32BIT=ON | | fv3 | 142 >>>>>>> upstream/develop
> > > this is what I got when merging @climbfuji branch, I used the develop one in my push, I guess Dom didn't get chance to test for this before he created PR to my branch
> > 
> > 
> > @jiandewang @climbfuji I think we can revert the line 139. geae is down today but I tested ok with the reverted line 139.
> 
> Both Sam and Dom modified that compiling line, that's why I got conflict. @climbfuji can you confirm on what @jkbk2004 suggested in the above ?

@jkbk2004 after you revert the compiling line, all jobs under this block passed, right ?

jkbk2004 commented:
> > > > 138 <<<<<<< HEAD 139 COMPILE | -DAPP=ATM -DDEBUG=ON -DCCPP_SUITES=FV3_RAP,FV3_RAP_cires_ugwp,FV3_RAP_unified_ugwp,FV3_RAP_flake -D32BIT=ON | + wcoss_dell_p3 wcoss_cray | fv3 | 140 ======= 141 COMPILE | -DAPP=ATM -DDEBUG=ON -DCCPP_SUITES=FV3_RAP,FV3_RAP_cires_ugwp,FV3_RAP_unified_ugwp,FV3_RAP_flake,FV3_HRRR -D32BIT=ON | | fv3 | 142 >>>>>>> upstream/develop
> > > > this is what I got when merging @climbfuji branch, I used the develop one in my push, I guess Dom didn't get chance to test for this before he created PR to my branch
> > > 
> > > 
> > > @jiandewang @climbfuji I think we can revert the line 139. geae is down today but I tested ok with the reverted line 139.
> > 
> > 
> > Both Sam and Dom modified that compiling line, that's why I got conflict. @climbfuji can you confirm on what @jkbk2004 suggested in the above ?
> 
> @jkbk2004 after you revert the compiling line, all jobs under this block passed, right ?

Sure! I will update logs after reverting the line.

jiandewang commented:
> > > > > 138 <<<<<<< HEAD 139 COMPILE | -DAPP=ATM -DDEBUG=ON -DCCPP_SUITES=FV3_RAP,FV3_RAP_cires_ugwp,FV3_RAP_unified_ugwp,FV3_RAP_flake -D32BIT=ON | + wcoss_dell_p3 wcoss_cray | fv3 | 140 ======= 141 COMPILE | -DAPP=ATM -DDEBUG=ON -DCCPP_SUITES=FV3_RAP,FV3_RAP_cires_ugwp,FV3_RAP_unified_ugwp,FV3_RAP_flake,FV3_HRRR -D32BIT=ON | | fv3 | 142 >>>>>>> upstream/develop
> > > > > this is what I got when merging @climbfuji branch, I used the develop one in my push, I guess Dom didn't get chance to test for this before he created PR to my branch
> > > > 
> > > > 
> > > > @jiandewang @climbfuji I think we can revert the line 139. geae is down today but I tested ok with the reverted line 139.
> > > 
> > > 
> > > Both Sam and Dom modified that compiling line, that's why I got conflict. @climbfuji can you confirm on what @jkbk2004 suggested in the above ?
> > 
> > 
> > @jkbk2004 after you revert the compiling line, all jobs under this block passed, right ?
> 
> Sure! I will update logs after reverting the line.

@jkbk2004  I will revert that line and commit to my branch once I got confirmation from you.

BrianCurtis-NOAA commented:
I would also recommend merging @climbfuji PR in and keeping `FV3_HRRR` at the end of the `CCPP_SUITES` line.

jkbk2004 commented:
> > > > > > 138 <<<<<<< HEAD 139 COMPILE | -DAPP=ATM -DDEBUG=ON -DCCPP_SUITES=FV3_RAP,FV3_RAP_cires_ugwp,FV3_RAP_unified_ugwp,FV3_RAP_flake -D32BIT=ON | + wcoss_dell_p3 wcoss_cray | fv3 | 140 ======= 141 COMPILE | -DAPP=ATM -DDEBUG=ON -DCCPP_SUITES=FV3_RAP,FV3_RAP_cires_ugwp,FV3_RAP_unified_ugwp,FV3_RAP_flake,FV3_HRRR -D32BIT=ON | | fv3 | 142 >>>>>>> upstream/develop
> > > > > > this is what I got when merging @climbfuji branch, I used the develop one in my push, I guess Dom didn't get chance to test for this before he created PR to my branch
> > > > > 
> > > > > 
> > > > > @jiandewang @climbfuji I think we can revert the line 139. geae is down today but I tested ok with the reverted line 139.
> > > > 
> > > > 
> > > > Both Sam and Dom modified that compiling line, that's why I got conflict. @climbfuji can you confirm on what @jkbk2004 suggested in the above ?
> > > 
> > > 
> > > @jkbk2004 after you revert the compiling line, all jobs under this block passed, right ?
> > 
> > 
> > Sure! I will update logs after reverting the line.
> 
> @jkbk2004 I will revert that line and commit to my branch once I got confirmation from you.

Sure, Jiande! Please, go ahead to revert the line. Can you consider Brian's comment below as well?

jiandewang commented:
> I would also recommend merging @climbfuji PR in and keeping `FV3_HRRR` at the end of the `CCPP_SUITES` line.

@BrianCurtis-NOAA that is what I did last Friday. I merged @climbfuji branch but used what's in UFS header for that line (with FV3_HRRR). I believe in @jkbk2004's successful jobs he removed FV3_HRRR, let's wait for his confirmation.

jiandewang commented:
@jkbk2004 @BrianCurtis-NOAA I am a bit confused, this is what we are having now:
COMPILE | -DAPP=ATM -DDEBUG=ON -DCCPP_SUITES=FV3_RAP,FV3_RAP_cires_ugwp,FV3_RAP_unified_ugwp,FV3_RAP_flake,FV3_HRRR -D32BIT=ON

shall I simply removed FV3_HRRR or do something else ?
@jkbk2004 : in your successful jobs, did you simply removed "FV3_HRRR" ?

jkbk2004 commented:
> @jkbk2004 @BrianCurtis-NOAA I am a bit confused, this is what we are having now: COMPILE | -DAPP=ATM -DDEBUG=ON -DCCPP_SUITES=FV3_RAP,FV3_RAP_cires_ugwp,FV3_RAP_unified_ugwp,FV3_RAP_flake,FV3_HRRR -D32BIT=ON
> 
> shall I simply removed FV3_HRRR or do something else ? @jkbk2004 : in your successful jobs, did you simply removed "FV3_HRRR" ?

![rt-conf-diff-1256](https://user-images.githubusercontent.com/38663008/174838519-bef3f28f-0218-4085-aedf-eb839b5188b4.png)

jkbk2004 commented:
This is only change I made. It ran ok. I am btw meetings today.

BrianCurtis-NOAA commented:
> I checked rap_noah_debug failed case. Develop branch runs ok but it crashes with the changes with the PR:
> 111:An error occurred in ccpp_physics_init
> 111:Invalid suite FV3_RAP_noah

This is the only mention of the error we saw from RDHPCS machines. I don't know why we are trying to remove FV3_HRRR then.

jkbk2004 commented:
> > I checked rap_noah_debug failed case. Develop branch runs ok but it crashes with the changes with the PR:
> > 111:An error occurred in ccpp_physics_init
> > 111:Invalid suite FV3_RAP_noah
> 
> This is the only mention of the error we saw from RDHPCS machines. I don't know why we are trying to remove FV3_HRRR then.

@jiandewang @BrianCurtis-NOAA Brian, you are right. Let me make clear about the exact line number. its 150 on Jiande's current branch. I think line number 150 is all we need to revert, right?

jiandewang commented:
> > > I checked rap_noah_debug failed case. Develop branch runs ok but it crashes with the changes with the PR:
> > > 111:An error occurred in ccpp_physics_init
> > > 111:Invalid suite FV3_RAP_noah
> > 
> > 
> > This is the only mention of the error we saw from RDHPCS machines. I don't know why we are trying to remove FV3_HRRR then.
> 
> @jiandewang @BrianCurtis-NOAA Brian, you are right. Let me make clear about the exact line number. its 150 on Jiande's current branch. I think line number 150 is all we need to revert, right?

@jkbk2004 I think you are right, although merging conflict happened on line 138, the failure is with line 150. Can you share me your directory as I can't tell what you changed from your screen shot ? What I can see is you removed wcoss_*

jkbk2004 commented:
> @jkbk2004 @BrianCurtis-NOAA I am a bit confused, this is what we are having now: COMPILE | -DAPP=ATM -DDEBUG=ON -DCCPP_SUITES=FV3_RAP,FV3_RAP_cires_ugwp,FV3_RAP_unified_ugwp,FV3_RAP_flake,FV3_HRRR -D32BIT=ON
> 
> shall I simply removed FV3_HRRR or do something else ? @jkbk2004 : in your successful jobs, did you simply removed "FV3_HRRR" ?



> @jkbk2004 @BrianCurtis-NOAA I am a bit confused, this is what we are having now: COMPILE | -DAPP=ATM -DDEBUG=ON -DCCPP_SUITES=FV3_RAP,FV3_RAP_cires_ugwp,FV3_RAP_unified_ugwp,FV3_RAP_flake,FV3_HRRR -D32BIT=ON
> 
> shall I simply removed FV3_HRRR or do something else ? @jkbk2004 : in your successful jobs, did you simply removed "FV3_HRRR" ?

No, I didn't remove FV3_HRRR.

jiandewang commented:
@jkbk2004 I think previously we had miscommunication on the exact line where the issue happened. Now let's forget line 138 but focus on line 150. You told us that your jobs passed after you modified line 150. My question is what did you changes here ? Can you tell me the directory so that I can compare ? What I saw is you removed wcoss machine in this line but problem happened on R&D machine.

jkbk2004 commented:
> @jkbk2004 I think previously we had miscommunication on the exact line where the issue happened. Now let's forget line 138 but focus on line 150. You told us that your jobs passed after you modified line 150. My question is what did you changes here ? Can you tell me the directory so that I can compare ? What I saw is you removed wcoss machine in this line but problem happened on R&D machine.

@jiandewang My full test on Cheyenne. We need to remove + wcoss/dell/cray on line 150. That's all we need.

BrianCurtis-NOAA commented:
@jiandewang @jkbk2004 On WCOSS_Cray all tests pass. On WCOSS_Dell_P3, the following tests failed: control_debug, control_ras_debug, rap_lndp_debug, hafs_regional_atm

jiandewang commented:
> @jiandewang @jkbk2004 On WCOSS_Cray all tests pass. On WCOSS_Dell_P3, the following tests failed: control_debug, control_ras_debug, rap_lndp_debug, hafs_regional_atm

I will removed wcoss from line 150 around 1:30pm, meeting now, still not sure that's what we need

jkbk2004 commented on tests/rt.conf:
```diff
@@ -145,7 +147,7 @@ RUN     | rap_lndp_debug
 RUN     | rap_progcld_thompson_debug                                                                                              |                                         | fv3 |
 
-COMPILE | -DAPP=ATM -DDEBUG=ON -DCCPP_SUITES=FV3_RAP_noah,FV3_RAP_RRTMGP,FV3_RAP_sfcdiff,FV3_RAP_noah_sfcdiff_cires_ugwp,FV3_RRFS_v1beta -D32BIT=ON |                       | fv3 |
+COMPILE | -DAPP=ATM -DDEBUG=ON -DCCPP_SUITES=FV3_RAP_noah,FV3_RAP_RRTMGP,FV3_RAP_sfcdiff,FV3_RAP_noah_sfcdiff_cires_ugwp,FV3_RRFS_v1beta -D32BIT=ON | + wcoss_dell_p3 wcoss_cray | fv3 |
```
@jiandewang I still see wcoss dell/cray at line 150.

jiandewang replied:
just updated my branch

BrianCurtis-NOAA commented:
> > @jkbk2004 I think previously we had miscommunication on the exact line where the issue happened. Now let's forget line 138 but focus on line 150. You told us that your jobs passed after you modified line 150. My question is what did you changes here ? Can you tell me the directory so that I can compare ? What I saw is you removed wcoss machine in this line but problem happened on R&D machine.
> 
> @jiandewang My full test on Cheyenne. We need to remove + wcoss/dell/cray on line 150. That's all we need.

If you were working on Cheyenne, why are you removing those restrictions to only run on WCOSS machines?

jkbk2004 commented:
> > > @jkbk2004 I think previously we had miscommunication on the exact line where the issue happened. Now let's forget line 138 but focus on line 150. You told us that your jobs passed after you modified line 150. My question is what did you changes here ? Can you tell me the directory so that I can compare ? What I saw is you removed wcoss machine in this line but problem happened on R&D machine.
> > 
> > 
> > @jiandewang My full test on Cheyenne. We need to remove + wcoss/dell/cray on line 150. That's all we need.
> 
> If you were working on Cheyenne, why are you removing those restrictions to only run on WCOSS machines?

@BrianCurtis-NOAA That's was my question this morning. why ccpp_init complains with + wcoss/dell/cray there? parsing issue? And without that + wcoss options, it ran ok. Same symptom is happening on all machines.

jiandewang commented:
@BrianCurtis-NOAA can you check line 150 of rt.conf for me ? 
in my testing run, I got error
echo 'MACHINES=|fv3|'
+ die 'MACHINES spec must be either an empty string or start with either '\''+'\'' or '\''-'\

Am I have any typo here ?

BrianCurtis-NOAA commented:
> @BrianCurtis-NOAA can you check line 150 of rt.conf for me ? in my testing run, I got error echo 'MACHINES=|fv3|'
> 
> * die 'MACHINES spec must be either an empty string or start with either '''+''' or '''-'\
> 
> Am I have any typo here ?

You have an extra vertical line.

jiandewang commented:
I just made a test run for "rap_noah_debug", it did passed, just don't understand why 
@jkbk2004 and @BrianCurtis-NOAA can you put RT label back to on of the R&D machine ?

jkbk2004 commented:
@jiandewang I will update logs that I have.

DeniseWorthen commented:
@jkbk2004 Were your cheyenne logs generated with the final PR commit ([dabcf03](https://github.com/ufs-community/ufs-weather-model/pull/1256/commits/dabcf03b3529597efd91d3d5cfb30b90f1b8a00b))

jkbk2004 commented:
> @jkbk2004 Were your cheyenne logs generated with the final PR commit ([dabcf03](https://github.com/ufs-community/ufs-weather-model/pull/1256/commits/dabcf03b3529597efd91d3d5cfb30b90f1b8a00b))

@DeniseWorthen The logs were created with the change needed. But I will add labels to make sure everything ok.

jiandewang commented:
> @jkbk2004 Were your cheyenne logs generated with the final PR commit ([dabcf03](https://github.com/ufs-community/ufs-weather-model/pull/1256/commits/dabcf03b3529597efd91d3d5cfb30b90f1b8a00b))

I think for safety purpose, let's start all jobs from scratch

jkbk2004 commented:
@jiandewang sure! I am re-running on cheyenne as well now.

BrianCurtis-NOAA commented:
***Automated RT Failure Notification***
Machine: jet
Compiler: intel
Job: RT
[RT] Repo location: /lfs4/HFIP/h-nems/emc.nemspara/autort/pr/959824692/20220621183017/ufs-weather-model
[RT] Error: Test control_c384gdas 023 failed in run_test failed 
Please make changes and add the following label back: jet-intel-RT

jkbk2004 commented:
@jiandewang ready to merge in. Can we start to merge in dependencies?

DeniseWorthen commented:
@jiandewang CMEPS is merged. New hash is [94c5e8d](https://github.com/NOAA-EMC/CMEPS/commit/94c5e8d7f204f95a3b93e1f6533a205c3ede7002)

DeniseWorthen commented:
@jiandewang I don't see the correct CMEPS hash in your branch. Top of emc/develop is [94c5e8d](https://github.com/NOAA-EMC/CMEPS/commit/94c5e8d7f204f95a3b93e1f6533a205c3ede7002)

jiandewang commented:
> @jiandewang I don't see the correct CMEPS hash in your branch. Top of emc/develop is [94c5e8d](https://github.com/NOAA-EMC/CMEPS/commit/94c5e8d7f204f95a3b93e1f6533a205c3ede7002)

@DeniseWorthen you are right, I modified the .gitsubmodule but MOM6 and CMEPS are not pointing to the latest hash
all I did is modified that file and commit, is there any step I missed ?

DeniseWorthen commented:
@jiandewang Try 'git submodule sync' and don't forget to 'add CMEPS' and commit.

jiandewang commented:
MOM6 and CMEPS are pointing to dev/emc now

DeniseWorthen reviewed (approved):
Submodules are correct.

jiandewang commented:
@jkbk2004 @BrianCurtis-NOAA @DeniseWorthen @climbfuji thanks for everybody's work to have these 3 PR done in one shot.
